### DISCUSSION TITLE
Installation fails for mahotas on linux-aarch64

### DISCUSSION THREAD

**Discussion by ligaz** [python-poetry/poetry#5574] [Q&A] [answered]
Installing the [mahotas](https://github.com/luispedro/mahotas) library fails on ARM linux distros with:
```
  RuntimeError

  Unable to find installation candidates for mahotas (1.4.12)

  at ~/.local/share/pypoetry/venv/lib/python3.7/site-packages/poetry/installation/chooser.py:73 in choose_for
       69│             links.append(link)
       70│ 
       71│         if not links:
       72│             raise RuntimeError(
    →  73│                 "Unable to find installation candidates for {}".format(package)
       74│             )
       75│ 
```

Installation works with pip:
```sh
$ python3.7 -m pip install mahotas
```

Poetry has been installed with:
```sh
$ curl -sSL https://install.python-poetry.org | python3.7 -
```

Any ideas?

**Comment by clintonroy** [comment, marked as answer]
If you check out https://pypi.org/project/mahotas/#files you can see there's no source distribution and no arm build. Pip is probably downloading 1.4.11, which does have a source dist.

try poetry add mahotas==1.4.11

**Reply by ligaz** [reply]
Indeed that seems to be the case. Thanks for your reply! 

Do you think poetry should behave similarly (use a the latest version with source distribution if there are no pre-built binaries for the current architecture) in such cases?

**Reply by abn** [reply]
I would say not. Because when the lockfile is resolved, it is done based on version constraints. We do not know in advance what platforms need to be supported or if the version intentionally does not distribute sdists. 

If we were to enable this configuration for projects then yes we can do it.

That said, this only really applies when using `poetry install`. You can still pip install your project and the frontend can choose what it wants to do if the dependency constraint is met.

**Comment by clintonroy** [comment]
I dunno to be honest. It's better in some regards, worse in others.
